src/api/forms/service.js

```diff
@@ -189,6 +189,56 @@ export async function updateDraftFormDefinition(formId, definition, author) {
   }
 }
 
+/**
+ * @param {string} formId - ID of the form
+ * @param {Partial<FormMetadataInput>} formUpdate - full JSON form definition
+ * @returns {Promise<string>}
+ */
+// TODO: author to be used to update updatedBy in next PR
+export async function updateFormMetadata(formId, formUpdate) {
+  logger.info(`Updating form metadata (draft) for form ID ${formId}`)
+
+  try {
+    // Get the form metadata from the db
+    const form = await getForm(formId)
+
+    if (form.live && formUpdate.title) {
+      throw Boom.badRequest(
+        `Field ${formUpdate.title} cannot be updated once the form has gone live`
+      )
+    }
+
+    const updatedSlug = slugify(formUpdate.title)
```

Comment:
We're creating a slug even though `formUpdate.title` might not be present.
How about:

```
const updatedForm = { ...formUpdate }

if (formUpdate.title) {
  updatedForm.slug = slugify(formUpdate.title)
}
```